### ISSUE TITLE
Wish list

### ISSUE THREAD

**Issue by SDAASDDASD** [gridap/Gridap.jl#214]
Hello!

can we have a wish list (numerical features / functions)for Gridap?

best regards!

**Comment by fverdugo** [comment]
Yes of course, in this issue for the moment. 

In any case, we have a more or less clear road map and we are quite busy implementing it. So, the wishes that are aligned with our road map are likely to be available in the future, but not otherwise.

**Comment by SDAASDDASD** [comment]
Hi @fverdugo,

thanks for your reply!

how about the following items for Gridap:

1.  Coupled transient problems, e.g.: coupled hydro-mechanical model.
2.  Parallel Processing via MPI & OpenMP
3.  Inelasticity(small/finite) problems

Thanks a lot in advance!
Best!

**Comment by santiagobadia** [comment]
Hi,

1 and 3 can be done with the current machinery. Multifield/multiphysics are working. We have not covered transient problems in the tutorials because we are not providing time integrators yet. But, e.g., implement a theta method in your driver is a matter of lines. To provide time integration is one of the planned next steps.

2 is at another level of complexity, especially for MPI. This is a project we want to endeavor _soon_ but it will take some time.

Regards

--Santi

**Comment by SDAASDDASD** [comment]
@santiagobadia ,

thank you for your answer!

1 and 3, is it possible that we can use the staggered scheme for Multifield problems:
 And automatic time stepping?

In addition, can we use the h adaptivity or hp adaptivity refinement in Gridap?

Best!

**Comment by santiagobadia** [comment]
A particular staggered scheme for a multifield problem is something that won't go to the `Gridap` library, it is very specific to the problem at hand. So, I would consider to develop it at the _driver_ level, probably combined with the time stepping tools to be provided in the near future.

`h` and `hp`-adaptivity is in our wish list too. But again, this is a major release. And in parallel, quite a challenge.

**Comment by SDAASDDASD** [comment]
for h and hp-adaptivity, can we add a third part toolbox **"p4est"** to Gridap in the near future? 
[https://github.com/cburstedde/p4est]

It is written in C++ .....

**Comment by santiagobadia** [comment]
Yes, sure! 
It would be nice to have a p4est wrapper for Julia first.
If you have the time, we can start a project on this.
I think that this wrapper should be in a separate `Gridap` package, like we have already done for some linear solvers and meshing tools.

**Comment by fverdugo** [comment]
Contributions to the Gridap project are welcome. However, before adding any code by external collaborators into the Gridap organization I would like to review this code to check that it fulfills our standards.

**Comment by santiagobadia** [comment]
We take note of the suggestions. I close the issue.
